### PR TITLE
Improved seletopclusts

### LINKED ISSUE #789: Add a way to rank clusters in `seletopclusts`
**Is your feature request related to a problem? Please describe.**
We do not know how are ranked clusters in `seletopclusts`

**Describe the solution you'd like**
- Add a proper documentation
- Add new way to select clusters, based on energetics or cluster size

### PR DESCRIPTION
**Pull request by VGPReys**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md) and that you comply with the following criteria:

- [X] You have sticked to Python. Please talk to us before adding other programming languages to HADDOCK3
- [ ] Your PR is about CNS
- [X] Your code is well documented: proper docstrings and explanatory comments for those tricky parts
- [X] You structured the code into small functions as much as possible. You can use classes if there is a (state) purpose
- [X] Your code follows our coding style
- [X] You wrote tests for the new code
- [X] `tox` tests pass. *Run `tox` command inside the repository folder*
- [X] `-test.cfg` examples execute without errors. *Inside `examples/` run `python run_tests.py -b`*
- [X] PR does not add any dependencies, unless permission granted by the HADDOCK team
- [X] PR does not break licensing
- [ ] Your PR is about writing documentation for already existing code :fire:
- [X] Your PR is about writing tests for already existing code :godmode:

---

This PR improves the `seletopclusts` module by:
- [x] Adding new `sortby` parameter, allowing to sort clusters based on `score` or `size`.
- [x] Emptying the `__init__.py` and place dedicated function in a separated script.
- [x] Adding tests for the module.
- [x] Adding a `seletopclust` example

Closes #789.

### CODE REVIEW

File: tests/test_module_seletopclusts.py [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+    select_top_clusts_models,
+    rank_clust_order,
+    size_clust_order,
+    sort_models,
+    write_selected_models,
+    )
+
+
+NB_CLUSTERS = 3
+MDL_RANK_INCREMENT = 5
```
Comment (mgiulini) [inline tests/test_module_seletopclusts.py:24]: what do these two variables represent?

File: src/haddock/modules/analysis/seletopclusts/defaults.yaml [hunk 2]
```diff
@@ -18,3 +18,16 @@ top_models:
...
   explevel: easy
+method:
+  default: score
+  type: string
+  minchars: 2
+  maxchars: 15
+  choice:
+    - score
+    - size
+  title: Method used to define `best cluster`.
```
Comment (mgiulini) [inline src/haddock/modules/analysis/seletopclusts/defaults.yaml:29]: for consistency with caprieval I would call this parameter "sortby" (see https://github.com/haddocking/haddock3/blob/cb4f1945ca7db4867426c79911f611a327a4ea00/src/haddock/modules/analysis/caprieval/defaults.yaml#L113C1-L113C7)

File: src/haddock/modules/analysis/seletopclusts/defaults.yaml [hunk 3]
```diff
@@ -18,3 +18,16 @@ top_models:
...
+  short: Best cluster can be defined based either on cluster models scores, or on cluster size. By default, 'score' is selected.
+  long: For cluster models scores, models scores are averaged giving the cluster rank and clusters are ordered according to their rank. For clustering by cluster size, biggest cluster size corresponds to higher rank. By default, 'score' is selected.
```
Comment (mgiulini) [inline src/haddock/modules/analysis/seletopclusts/defaults.yaml:31]:
```suggestion
  long: if the selection is done by 'score' the average score of the top (4) models of each cluster is used to define the cluster rank. When clustering by 'size', a bigger cluster size corresponds to a higher rank. By default, 'score' is selected.
```

**Review by mgiulini — APPROVED** [review]

File: tests/test_module_seletopclusts.py (hunk 1 above)
Comment (VGPReys) [reply tests/test_module_seletopclusts.py:24]: A way to standardize the counting in generation of test datasets and later tests.

File: src/haddock/modules/analysis/seletopclusts/defaults.yaml (hunk 2 above)
Comment (VGPReys) [reply src/haddock/modules/analysis/seletopclusts/defaults.yaml:29]: OK nice

**Pull request merged**